buildkite/bazelci.py

```diff
@@ -1285,6 +1304,54 @@ def execute_bazel_build(
         handle_bazel_failure(e, "build")
 
 
+def expand_test_target_patterns(bazel_binary, platform, test_targets):
+    if not any(t for t in test_targets if t.endswith(":all") or t.endswith("...")):
+        return test_targets
+
+    eprint("Resolving test targets via bazel query")
+    output = execute_command_and_get_output(
+        [bazel_binary]
+        + common_startup_flags(platform)
+        + [
+            "--nomaster_bazelrc",
+            "--bazelrc=/dev/null",
+            "query",
+            "tests(set({}))".format(" ".join("'{}'".format(t) for t in test_targets if t != "--")),
```

Comment:
I think this doesn't work for blacklisted tests, because `tests(set(...))` doesn't seem to understand the "-//my:target" syntax. I tried it locally like this:

```
$ bazel query "tests(set('//src/test/...' '-//src/test/shell/bazel:bazel_determinism_test'))" 2>/dev/null | grep determinism
//src/test/shell/bazel:bazel_determinism_test
```

However, if we slightly change the query, it seems to work fine:

```
$ bazel query "tests(set('//src/test/...')) except set('//src/test/shell/bazel:bazel_determinism_test')" 2>/dev/null | grep determinism
```

It wouldn't work if someone tries to exclude an entire test suite (e.g. `//src/test/shell:all_windows_tests`), because that's expanded by `tests(...)` before the `except set(...)` happens, but maybe that's still OK. I guess one usually only disables individual tests anyway.

WDYT?